File: go/tests/array_direct_test.go
package llml_test

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/zenbase-ai/llml/go/pkg/llml"
)

func TestDirectArray(t *testing.T) {
	result := llml.Sprintf([]any{"a", "b", "c"})
	expected := "<1>a</1>\n<2>b</2>\n<3>c</3>"
	assert.Equal(t, expected, result)
}

func TestDirectArrayWithMixedTypes(t *testing.T) {
	result := llml.Sprintf([]any{1, "hello", true})
	expected := "<1>1</1>\n<2>hello</2>\n<3>true</3>"
	assert.Equal(t, expected, result)
}

func TestDirectArrayWithObjects(t *testing.T) {
	result := llml.Sprintf([]any{
		map[string]any{"name": "Alice"},
		map[string]any{"name": "Bob"},
	})
	expected := "<1>\n  <1-name>Alice</1-name>\n</1>\n<2>\n  <2-name>Bob</2-name>\n</2>"
	assert.Equal(t, expected, result)
}

func TestDirectEmptyArray(t *testing.T) {
	result := llml.Sprintf([]any{})
	expected := ""
	assert.Equal(t, expected, result)
}

func TestDirectArrayWithIndentation(t *testing.T) {
	result := llml.Sprintf([]any{"a", "b"}, llml.Options{Indent: "  "})
	expected := "  <1>a</1>\n  <2>b</2>"
	assert.Equal(t, expected, result)
}

func TestDirectArrayWithPrefix(t *testing.T) {
	result := llml.Sprintf([]any{"a", "b"}, llml.Options{Prefix: "item"})
	expected := "<item-1>a</item-1>\n<item-2>b</item-2>"
	assert.Equal(t, expected, result)
}

File: py/tests/array_direct_test.py
from zenbase_llml import llml


def test_direct_array_with_strings():
    """Test direct array with string values."""
    result = llml(["a", "b", "c"])
    expected = "<1>a</1>\n<2>b</2>\n<3>c</3>"
    assert result == expected


def test_direct_array_with_mixed_types():
    """Test direct array with mixed types."""
    result = llml([1, "hello", True])
    expected = "<1>1</1>\n<2>hello</2>\n<3>True</3>"
    assert result == expected


def test_direct_array_with_objects():
    """Test direct array containing dictionaries (default non-strict mode)."""
    result = llml([{"name": "Alice"}, {"name": "Bob"}])
    expected = "<1><name>Alice</name></1>\n<2><name>Bob</name></2>"
    assert result == expected


def test_direct_empty_array():
    """Test direct empty array."""
    result = llml([])
    expected = ""
    assert result == expected


def test_direct_array_with_indentation():
    """Test direct array with indentation options."""
    # Note: New API doesn't support indentation options
    result = llml(["a", "b"])
    expected = "<1>a</1>\n<2>b</2>"
    assert result == expected


def test_direct_array_with_prefix():
    """Test direct array with prefix options."""
    # Note: New API doesn't support prefix options
    result = llml(["a", "b"])
    expected = "<1>a</1>\n<2>b</2>"
    assert result == expected


def test_direct_array_with_objects_strict_mode():
    """Test direct array containing dictionaries with strict mode enabled."""
    # Note: New API doesn't support strict mode options
    result = llml([{"name": "Alice"}, {"name": "Bob"}])
    expected = "<1><name>Alice</name></1>\n<2><name>Bob</name></2>"
    assert result == expected
